### ISSUE TITLE
Management module overrides set parameters for policies

### ISSUE THREAD

**Issue by TobiasKeitsch** [Azure/terraform-azurerm-caf-enterprise-scale#97]
# Problem statement
Tested with:
Terraform version: `0.14.10` and `0.15.3`
AzureRM provider version: `latest (2.58.0)`
Module version: `0.2.0` and `0.3.0`

Some ES policies, that are applied to the root with manually specified parameters, want to set the parameters to the default value originating from the management module, even if we don't deploy the management resources from the ES module. 
It doesn't matter if we apply our parameters in the PolicyAssignment, ArchetypeDefinition, or archetype_config_overrides, it still choose the settings from the management module.

A workaround that almost work all way, is to configure `configure_management_resources` with a mockup where you can configure the es module with this property:
```
configure_management_resources = {
    settings = {
      log_analytics = {
        enabled = true
        config = {
          retention_in_days                           = 30
          enable_monitoring_for_arc                   = true
          enable_monitoring_for_vm                    = true
          enable_monitoring_for_vmss                  = true
          enable_solution_for_agent_health_assessment = true
          enable_solution_for_anti_malware            = true
          enable_solution_for_azure_activity          = true
          enable_solution_for_change_tracking         = true
          enable_solution_for_service_map             = true
          enable_solution_for_sql_assessment          = true
          enable_solution_for_updates                 = true
          enable_solution_for_vm_insights             = true
          enable_sentinel                             = true
        }
      }
      security_center = {
        enabled = true
        config = {
          email_security_contact             = "security-contact@fabrikam.com"
          enable_defender_for_acr            = true
          enable_defender_for_app_services   = true
          enable_defender_for_arm            = true
          enable_defender_for_dns            = true
          enable_defender_for_key_vault      = true
          enable_defender_for_kubernetes     = true
          enable_defender_for_servers        = true
          enable_defender_for_sql_servers    = true
          enable_defender_for_sql_server_vms = true
          enable_defender_for_storage        = true
        }
      }
    }

    location = "westeurope"
    tags     = null
    advanced = {
      existing_resource_group_name                 = "custom-rg"
      existing_log_analytics_workspace_resource_id = "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/custom-rg/providers/Microsoft.OperationalInsights/workspaces/custom-la"
      existing_automation_account_resource_id      = "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/custom-rg/providers/Microsoft.OperationalInsights/workspaces/custom-aa"
      custom_settings_by_resource_type = {
        azurerm_automation_account = {
          name = "custom-aa"
        }
        azurerm_log_analytics_workspace = {
          name = "custom-la"
        }
      }
    }
  }
```

But there is still 1 hardcoded parameter in the management module for the policy Deploy-ASC-Defender which is:
`ascExportResourceGroupName          = "${local.root_id}-asc-export"`

This blocks an upgrade to the new versions, or creation of custom named resource group for that policy.

# Problem reproduction

Create a ES setup with the following config:
```
module "enterprise_scale" {
  source  = "Azure/caf-enterprise-scale/azurerm"
  version = "0.3.0"

  root_parent_id = "e30cb998-45a0-4111-b2fd-092fed41f457"
  #library_path     = "${path.root}/lib"
  default_location = "westeurope"
  root_id          = "es-test"
  root_name        = "es-test"
}
```
Run a terrafrom plan with the configuration and it displays the default values.
Example from the Deploy-ASC-Defender policyAssignment:
```
 # module.enterprise_scale.azurerm_policy_assignment.enterprise_scale["/providers/Microsoft.Management/managementGroups/es-test/providers/Microsoft.Authorization/policyAssignments/Deploy-ASC-Defender"] will be created
  + resource "azurerm_policy_assignment" "enterprise_scale" {
      + description          = "Deploy-ASC-Defender."
      + display_name         = "Deploy-ASC-Defender"
      + enforcement_mode     = false
      + id                   = (known after apply)
      + location             = "westeurope"
      + metadata             = (known after apply)
      + name                 = "Deploy-ASC-Defender"
      + not_scopes           = []
      + parameters           = jsonencode(
            {
              + ascExportResourceGroupLocation      = {
                  + value = "westeurope"
                }
              + ascExportResourceGroupName          = {
                  + value = "es-test-asc-export"
                }
              + emailSecurityContact                = {
                  + value = "security_contact@replace_me"
                }
              + logAnalytics                        = {
                  + value = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/es-test-mgmt/providers/Microsoft.OperationalInsights/workspaces/es-test-la"
                }
```

Adjust the ES module config with a archetype_config_overrides so the module config looks like this:

```
module "enterprise_scale" {
  source  = "Azure/caf-enterprise-scale/azurerm"
  version = "0.3.0"

  root_parent_id = "e30cb998-45a0-4111-b2fd-092fed41f457"
  #library_path     = "${path.root}/lib"
  default_location = "westeurope"
  root_id          = "es-test"
  root_name        = "es-test"

  archetype_config_overrides = {
    root = {
      archetype_id = "es_root"
      parameters = {
        Deploy-ASC-Defender = {
          emailSecurityContact                = "security-contact@fabrikam.com"
          logAnalytics                        = "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/custom-rg/providers/Microsoft.OperationalInsights/workspaces/custom-la",
          ascExportResourceGroupName          = "custom-rg"
          ascExportResourceGroupLocation      = "westeurope"
          pricingTierContainerRegistry        = "Standard"
          pricingTierAppServices              = "Standard"
          pricingTierArm                      = "Standard"
          pricingTierDns                      = "Standard"
          pricingTierKeyVaults                = "Standard"
          pricingTierKubernetesService        = "Standard"
          pricingTierVMs                      = "Standard"
          pricingTierSqlServers               = "Standard"
          pricingTierSqlServerVirtualMachines = "Standard"
          pricingTierStorageAccounts          = "Standard"
        }
      }
      access_control = {}
    }
  }
}
```
Run a terrafrom plan with the new configuration and it still displays the default values, instead of the ones that should have been overridden of the archetype_config_overrides config in root.
Example from the Deploy-ASC-Defender policyAssignment after the override:
```
 # module.enterprise_scale.azurerm_policy_assignment.enterprise_scale["/providers/Microsoft.Management/managementGroups/es-test/providers/Microsoft.Authorization/policyAssignments/Deploy-ASC-Defender"] will be created
  + resource "azurerm_policy_assignment" "enterprise_scale" {
      + description          = "Deploy-ASC-Defender."
      + display_name         = "Deploy-ASC-Defender"
      + enforcement_mode     = false
      + id                   = (known after apply)
      + location             = "westeurope"
      + metadata             = (known after apply)
      + name                 = "Deploy-ASC-Defender"
      + not_scopes           = []
      + parameters           = jsonencode(
            {
              + ascExportResourceGroupLocation      = {
                  + value = "westeurope"
                }
              + ascExportResourceGroupName          = {
                  + value = "es-test-asc-export"
                }
              + emailSecurityContact                = {
                  + value = "security_contact@replace_me"
                }
              + logAnalytics                        = {
                  + value = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/es-test-mgmt/providers/Microsoft.OperationalInsights/workspaces/es-test-la"
                }
```
Which is the same result and means that the `archetype_config_overrides` in the es module configuration is ignored.

# Expected behavior
If ES variable `deploy_management_resources` is set to false (or not set), the archetype_config_overrides from the management module should not be applied.

We should be able to use the ES policies used in the management module and set the parameters for the policies with either PolicyAssignment, ArchetypeDefinition, or archetype_config_overrides, without the need to configure the management module.
* Deploy-ASC-Defender
* Deploy-LX-Arc-Monitoring 
* Deploy-VM-Monitoring 
* Deploy-VMSS-Monitoring 
* Deploy-WS-Arc-Monitoring 
* Deploy-AzActivity-Log 
* Deploy-Resource-Diag

**Comment by gurrish** [comment]
We're seeing this issue for a customer too.

**Comment by krowlandson** [comment]
Great catch, and thank you for the feedback. I'll look into this ASAP as understand the issue this causes.

**Comment by TobiasKeitsch** [comment]
I noticed that, but I have not been using that setting yet. 
I think it would be part of the same kind of issue, since its enforcing settings from the module even if you don't use the module.

**Comment by krowlandson** [comment]
Closing issue as resolved by #98
